From cc2bf720bd82d71c87168b5bb6feb71ffa99c45a Mon Sep 17 00:00:00 2001
From: "Brint E. Kriebel" <bekit@cyngn.com>
Date: Sun, 25 Mar 2018 14:49:30 +0200
Subject: [PATCH 3/4] releasetools: Store the build.prop file in the OTA zip

This file is often used to read information about the update contained
in the OTA. Place it in the update so it can be used by scripts.

The file is not added to the updater-script, so it will not be placed
onto the actual system.

[GabrieleM: Adapt for A/B OTA packages]

Change-Id: I1a25a7974a5586192cf7bf026d918fc14e39714c
---
 tools/releasetools/ota_from_target_files.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/tools/releasetools/ota_from_target_files.py b/tools/releasetools/ota_from_target_files.py
index d02261ac6..326d4002b 100755
--- a/tools/releasetools/ota_from_target_files.py
+++ b/tools/releasetools/ota_from_target_files.py
@@ -1782,6 +1782,11 @@ def WriteABOTAPackageWithBrilloScript(target_file, output_file,
     secondary_payload.Sign(payload_signer)
     secondary_payload.WriteToZip(output_zip)
 
+  target_zip = zipfile.ZipFile(target_file, "r")
+  common.ZipWriteStr(output_zip, "system/build.prop",
+                     target_zip.read("SYSTEM/build.prop"))
+  common.ZipClose(target_zip)
+
   # If dm-verity is supported for the device, copy contents of care_map
   # into A/B OTA package.
   target_zip = zipfile.ZipFile(target_file, "r")
-- 
2.17.1

